pythonopensubtitles/opensubtitles.py

```diff
@@ -104,9 +109,99 @@ def detect_language(self):
+        :param filenames: optional dictionary with preferred file names;
+                          for keys use subtitle file IDs (as strings),
+                          for values use file names (incl. file extensions)
+        :param ext: one of: srt, sub, txt, ssa, smi, mpi, tmp;
+                    only applicable if only subtitle file IDs are provided
+        :param directory: path to preferred dir to which to write files
+        """
+        successful = {}
+        if filenames:
+            subfile_ids = list(filenames.keys())
+
+        # OpenSubtitles will accept a maximum of 20 IDs for download
+        if len(subfile_ids) > 20:
+            print("Cannot download more than 20 files at once.",
+                  file=sys.stderr)
+            subfile_ids = subfile_ids[:20]
+
+        self.data = self.xmlrpc.DownloadSubtitles(self.token, subfile_ids)
+        encoded_data = self._get_from_data_or_none('data')
+
+        if encoded_data:
+            for item in encoded_data:
+                subfile_id = item['idsubtitlefile']
+
+                try:
+                    decoded_data = zlib.decompress(
+                        base64.b64decode(item['data']),
+                        16 + zlib.MAX_WBITS).decode('utf-8')
+                except UnicodeDecodeError as e:
+                    try:
+                        decoded_data = zlib.decompress(
+                            base64.b64decode(item['data']),
+                            16 + zlib.MAX_WBITS).decode('latin1')
+                    except Exception as e:
+                        print("An error occurred during file decoding.",
+                              file=sys.stderr)
+                        print(e)
+                        continue
+                except Exception as e:
+                    print("An error occurred during file decoding.",
+                          file=sys.stderr)
+                    print(e)
+                    continue
+
+                if filenames:
+                    try:
+                        fpath = filenames[subfile_id]
+                    except Exception as e:
+                        print("'filenames' dictionary is not "
+                              "properly formatted. Exiting.", file=sys.stderr)
+                        sys.exit(1)
+                else:
+                    fpath = subfile_id + '.' + ext
+
+                if directory:
+                    fpath = os.path.join(directory, fpath)
+
+                try:
+                    with open(fpath, 'w') as f:
+                        f.write(decoded_data)
+                    successful[subfile_id] = fpath
+                except IOError as e:
+                    print(e)
+                except Exception as e:
+                    print("There was an error writing the subtitle data "
+                          "for file {}.".format(fpath),
+                          file=sys.stderr)
+                    print(e)
+
+        if successful:
```

Comment:
You can just `return successful or None`

Reply:
Ah, yeah, that would be analoguous to the returns in other functions. I always use "split" `if`-`else` constructs, I guess because I find them more readable (the one time I didn't it was in connection with testing, which I've generally done little of, and I remember getting confused again when I revisited the code :P). Will change.